receiver/opencensusreceiver/opencensus_test.go

```diff
@@ -179,6 +179,36 @@ func TestGrpcGatewayCors_endToEnd(t *testing.T) {
 	verifyCorsResp(t, url, "disallowed-origin.com", 200, false)
 }
 
+func TestMetricsGrpcGatewayCors_endToEnd(t *testing.T) {
+	addr := ":35991"
+	corsOrigins := []string{"allowed-*.com"}
+
+	sink := new(exportertest.SinkMetricsExporter)
+	ocr, err := New(addr, nil, sink, WithCorsOrigins(corsOrigins))
+	if err != nil {
+		t.Fatalf("Failed to create metrics receiver: %v", err)
+	}
+	defer ocr.StopMetricsReception()
+
+	mh := receivertest.NewMockHost()
+	go func() {
+		if err := ocr.StartMetricsReception(mh); err != nil {
+			t.Fatalf("Failed to start metrics receiver: %v", err)
+		}
+	}()
+
+	// Wait for the servers to start
+	<-time.After(10 * time.Millisecond)
```

Comment:
Looks fragile to me. This can be unstable if the execution is slow. Is there a better way to wait for servers to start for sure?

Reply:
For now the only return value of `Start*Reception` is `error` so it's hard to determine. I left a TODO here.

Reply:
It is hard for the interface to provide a good signal for that (ie.: typically they will launch a goroutine that will have a blocking call), so I think we will have to resort to something checking if the endpoint is up. That said for grpc the connection will do some retries under the hood so this code should be fine (famous last words :))